src/t_hash.c

```diff
@@ -2030,3 +2078,134 @@ void hrandfieldCommand(client *c) {
     hashTypeRandomElement(hash, hashTypeLength(hash), &ele, NULL);
     hashReplyFromListpackEntry(c, &ele);
 }
+
+/* Context structure for tracking expiry operations on hash fields. */
+typedef struct {
+    robj *key;               /* the hash object */
+    unsigned long n_entries; /* number of entries processed */
+    void **entries;          /* array of expired entries to replicate and free later */
+} expiryContext;
+
+/* Add an entry to the expiryContext list of processed entries. */
+static void ctxAddEntry(expiryContext *ctx, void *e) {
```

Comment:
seems like this function is just an overhead, I would remove it. If you do want to support a generic context, you should probably pass the Add function as a function pointer to the existing context

Reply:
The original idea behind ctxAddEntry() was encapsulation. But I agree this function adds unnecessary overhead for a one-liner used in a single place. I've removed it.